### PR TITLE
Whitelist

### PR DESCRIPTION
**Pull request by mswiszcz**
```ruby
require 'attributed_object'

class TestClass
  include AttributedObject::Strict

  CONST_WORLD = 'world'.freeze
  CONST_MARS = 'mars'.freeze

  attribute :planet, :string, whitelist: [CONST_WORLD, CONST_MARS]
end

TestClass.new(planet: TestClass::CONST_WORLD) # => Object instance with .planet = 'world'
TestClass.new(planet: 'sun') # => AttributedObject::DisallowedValueError
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,13 +70,22 @@ MyAttributedObject.new(
 SimpleFoo.new(bar: 12) == SimpleFoo.new(bar: 12)
 ```
 
+### Whitelist
+```ruby
+class MyAttributedObject
+  include AttributedObject::Strict
+
+  attribute :planet, :string, whitelist: ['Earth', 'Mars']
```
Comment (jgroeneveld) [inline README.md]: imho combine the documentation with
disallow and also make naming similar. maybe `only_allow:` ?

https://github.com/mswiszcz/attributed_object/blob/06db193d1e2533fa38ca01682894ac321a88bae5/README.md#disallow-nil

File: spec/attributed_object_spec.rb [hunk 2]
```diff
@@ -26,6 +26,15 @@ def self.count
     end
   end
 
+  class AttributedObjectWithWhitelist
+    include AttributedObject::Strict
+
+    PLANET_EARTH = 'earth'.freeze
+    PLANET_MARS = 'earth'.freeze
```
Comment (jgroeneveld) [inline spec/attributed_object_spec.rb]: type, i assume you mean mars?

**Review by jgroeneveld — CHANGES_REQUESTED** [review]

File: spec/attributed_object_spec.rb (hunk 2 above)
Comment (mswiszcz) [reply spec/attributed_object_spec.rb]: oops 😨

File: README.md (hunk 1 above)
Comment (mswiszcz) [reply README.md]: I actually disliked using only_allow, solely because disallow does not expect array but an exact value, that's why I went for "*list"

Comment (jgroeneveld) [reply README.md]: yes i had the same thought.. just `only`? for some reason whitelist feels a bit out of place in this dsl.. no idea why. you decide

Comment (mswiszcz) [reply README.md]: I like whitelists and blacklists, I would keep it as it is

**Pull request merged**